web/src/components/dashboard/HideableTasks.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { getMergedConfig } from "@/contexts/configContext";
+import { generatePreferences, generateProfileData, generateTask, generateUserData } from "@/test/factories";
+import { randomElementFromArray } from "@/test/helpers";
+import { useMockRoadmap } from "@/test/mock/mockUseRoadmap";
+import { useMockUserData } from "@/test/mock/mockUseUserData";
+import { setupStatefulUserDataContext, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import { OperatingPhases } from "@businessnjgovnavigator/shared/operatingPhase";
+import { UserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { HideableTasks } from "./HideableTasks";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+
+const Config = getMergedConfig();
+
+describe("HideableTasks", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupStatefulUserDataContext();
+    useMockRoadmap({});
+  });
+
+  const renderHideableTask = (initialUserData?: UserData) =>
+    render(
+      <WithStatefulUserData initialUserData={initialUserData}>
+        <HideableTasks />
+      </WithStatefulUserData>
+    );
+
+  it("does not render HideableTasks for operating phases that don't display HideableRoadmapTasks", () => {
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks !== true)
+    );
+    useMockUserData({ profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }) });
+    render(<HideableTasks />);
+
+    expect(screen.queryByText(Config.dashboardDefaults.upAndRunningTaskHeader)).not.toBeInTheDocument();
+  });
+
+  it("renders HideableTasks for operating phases that display HideableRoadmapTasks", () => {
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks === true)
+    );
+    useMockUserData({ profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }) });
+    render(<HideableTasks />);
+
+    expect(screen.getByText(Config.dashboardDefaults.upAndRunningTaskHeader)).toBeInTheDocument();
+  });
+
+  it("displays number of hidden tasks when hide toggle is clicked", () => {
+    useMockRoadmap({
+      tasks: [generateTask({}), generateTask({})],
+    });
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks === true)
+    );
+
+    const userData = generateUserData({
+      profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }),
+      preferences: generatePreferences({ isHideableRoadmapVisible: true }),
+    });
+    renderHideableTask(userData);
+
+    fireEvent.click(screen.getByText(Config.dashboardDefaults.hideTaskText));
+
+    expect(screen.getByText("2 Hidden Tasks")).toBeInTheDocument();
```

Comment:
we should reference config here instead of hardcoding the string in case the text gets changed by the content team